src/Http/Http.Extensions/src/ProblemDetailsJsonOptionsSetup.cs

```diff
@@ -11,15 +11,10 @@ internal sealed class ProblemDetailsJsonOptionsSetup : IPostConfigureOptions<Jso
 {
     public void PostConfigure(string? name, JsonOptions options)
     {
-        if (options.SerializerOptions.TypeInfoResolver is not null)
+        if (options.SerializerOptions.TypeInfoResolver is not null && options.SerializerOptions.TypeInfoResolver is not DefaultJsonTypeInfoResolver)
```

Comment:
This isn't making sense to me. Why don't we always want to use this context? Also changes like this that aren't obvious need to big comment 😄

Reply:
I can add a 😂 "huge" comment when we decided what is the right thing to do. As I mentioned `&& options.SerializerOptions.TypeInfoResolver is not DefaultJsonTypeInfoResolver` can be easily removed, however, it will be useless, unless we change how the `AddContext` works (to `prepend` instead of `append`).

Reply:
> unless we change how the `AddContext` works (to `prepend` instead of `append`)

After a conversation with @eiriktsarpalis, changing it doesn't make sense. As an example:

```c#
options.AddContext<Foo>();
options.AddContext<Bar>();
options.AddContext<Baz>();
```
Should be equivalent to writing:

``` c#
options.TypeInfoResolver = JsonTypeInfoResolver.Combine(Foo.Default, Bar.Default, Baz.Default);
```

Reply:
The problem seems like the JSON fallback needs to be applied *BEFORE* the first class to AddContext to make everything work in all cases. Is that right? We don't want to override the reflection resolver if it'll eventually be newed up lazily by the the JSON library.

Reply:
> JSON fallback needs to be applied _BEFORE_ the first class to AddContext to make

After. but everything else is correct. I believe we should keep the `options.SerializerOptions.TypeInfoResolver is not null` to avoid breaking an app due to our internal context but I am less concerned about `options.SerializerOptions.TypeInfoResolver is not DefaultJsonTypeInfoResolver` since it will not make any difference, what do you think?

Reply:
I tend to agree with David that the source generated context could go first even in cases where you have reflection enabled. It would save you a relatively small amount of initialization costs, particularly around JITing all the dynamic methods that `ProblemDetails` requires. But maybe not a huge win if everything else is dynamically generated.

You wouldn't need to change the semantics of `AddContext` to achieve that though, you could simply use the `Combine` method to prepend whatever resolver you like to user configuration.